diff -up mariadb-11.1.2/support-files/mariadb.logrotate.sh.1~ mariadb-11.1.2/support-files/mariadb.logrotate.sh
--- mariadb-11.1.2/support-files/mariadb.logrotate.sh.1~	2023-08-22 11:51:48.811510712 +0200
+++ mariadb-11.1.2/support-files/mariadb.logrotate.sh	2023-08-22 11:52:21.833837663 +0200
@@ -6,7 +6,8 @@
 # Read https://mariadb.com/kb/en/error-log/ to learn more about logging and
 # https://mariadb.com/kb/en/rotating-logs-on-unix-and-linux/ about rotating logs.
 
-@localstatedir@/mysqld.log @localstatedir@/mariadb.log @logdir@/*.log {
+@LOG_LOCATION@ @localstatedir@/mariadb.log @logdir@/*.log {
+  create 600 mysql mysql
 
   # Depends on a mysql@localhost unix_socket authenticated user with RELOAD privilege
   @su_user@
