diff -up mariadb-11.7.2/storage/innobase/include/trx0trx.h.omv~ mariadb-11.7.2/storage/innobase/include/trx0trx.h
--- mariadb-11.7.2/storage/innobase/include/trx0trx.h.omv~	2025-04-19 15:37:26.798341291 +0200
+++ mariadb-11.7.2/storage/innobase/include/trx0trx.h	2025-04-19 15:38:24.183688439 +0200
@@ -671,14 +671,18 @@ public:
   {
     ut_ad(!mutex_is_owner());
     mutex.wr_lock();
+#ifdef UNIV_DEBUG
     assert(!mutex_owner.exchange(pthread_self(),
                                  std::memory_order_relaxed));
+#endif
   }
   /** Release the mutex */
   void mutex_unlock()
   {
+#ifdef UNIV_DEBUG
     assert(mutex_owner.exchange(0, std::memory_order_relaxed) ==
            pthread_self());
+#endif
     mutex.wr_unlock();
   }
 #ifndef SUX_LOCK_GENERIC
