--- mariadb-10.1.1/storage/tokudb/PerconaFT/util/dmt.cc.omv~	2014-11-06 12:13:50.313697027 +0100
+++ mariadb-10.1.1/storage/tokudb/PerconaFT/util/dmt.cc	2014-11-06 12:18:32.684166911 +0100
@@ -922,9 +922,7 @@ void dmt<dmtdata_t, dmtdataout_t, dmtwri
 
 template<typename dmtdata_t, typename dmtdataout_t, typename dmtwriter_t>
 void dmt<dmtdata_t, dmtdataout_t, dmtwriter_t>::copyout(uint32_t *const outlen, dmtdata_t *const out, const dmt_node *const n) {
-    if (outlen) {
-        *outlen = n->value_length;
-    }
+    *outlen = n->value_length;
     if (out) {
         *out = n->value;
     }
@@ -932,19 +930,13 @@ void dmt<dmtdata_t, dmtdataout_t, dmtwri
 
 template<typename dmtdata_t, typename dmtdataout_t, typename dmtwriter_t>
 void dmt<dmtdata_t, dmtdataout_t, dmtwriter_t>::copyout(uint32_t *const outlen, dmtdata_t **const out, dmt_node *const n) {
-    if (outlen) {
-        *outlen = n->value_length;
-    }
-    if (out) {
-        *out = &n->value;
-    }
+    *outlen = n->value_length;
+    *out = &n->value;
 }
 
 template<typename dmtdata_t, typename dmtdataout_t, typename dmtwriter_t>
 void dmt<dmtdata_t, dmtdataout_t, dmtwriter_t>::copyout(uint32_t *const outlen, dmtdata_t *const out, const uint32_t len, const dmtdata_t *const stored_value_ptr) {
-    if (outlen) {
-        *outlen = len;
-    }
+    *outlen = len;
     if (out) {
         *out = *stored_value_ptr;
     }
@@ -952,12 +944,8 @@ void dmt<dmtdata_t, dmtdataout_t, dmtwri
 
 template<typename dmtdata_t, typename dmtdataout_t, typename dmtwriter_t>
 void dmt<dmtdata_t, dmtdataout_t, dmtwriter_t>::copyout(uint32_t *const outlen, dmtdata_t **const out, const uint32_t len, dmtdata_t *const stored_value_ptr) {
-    if (outlen) {
-        *outlen = len;
-    }
-    if (out) {
-        *out = stored_value_ptr;
-    }
+    *outlen = len;
+    *out = stored_value_ptr;
 }
 
 template<typename dmtdata_t, typename dmtdataout_t, typename dmtwriter_t>
