diff -up mariadb-12.2.2/cmake/install_layout.cmake.omv~ mariadb-12.2.2/cmake/install_layout.cmake
--- mariadb-12.2.2/cmake/install_layout.cmake.omv~	2026-02-18 17:05:58.258583976 +0100
+++ mariadb-12.2.2/cmake/install_layout.cmake	2026-02-18 17:10:49.512454172 +0100
@@ -157,13 +157,13 @@ SET(INSTALL_MYSQLTESTDIR_RPM
 SET(INSTALL_SQLBENCHDIR_RPM             "")
-SET(INSTALL_SUPPORTFILESDIR_RPM         "share/mariadb")
+SET(INSTALL_SUPPORTFILESDIR_RPM         "share/mysql")
 #
-SET(INSTALL_MYSQLDATADIR_RPM            "/var/lib/mysql")
+SET(INSTALL_MYSQLDATADIR_RPM            "/srv/mysql")
 
-SET(INSTALL_UNIX_ADDRDIR_RPM            "${INSTALL_MYSQLDATADIR_RPM}/mysql.sock")
 SET(INSTALL_SYSTEMD_UNITDIR_RPM         "/usr/lib/systemd/system")
 SET(INSTALL_SYSTEMD_SYSUSERSDIR_RPM     "/usr/lib/sysusers.d")
 SET(INSTALL_SYSTEMD_TMPFILESDIR_RPM     "/usr/lib/tmpfiles.d")
-SET(INSTALL_RUNDATADIR_RPM              "/run/mariadb")
+SET(INSTALL_RUNDATADIR_RPM              "/run/mysqld")
+SET(INSTALL_UNIX_ADDRDIR_RPM            "${INSTALL_RUNDATADIR_RPM}/mysql.sock")
 SET(INSTALL_PAMDIR_RPM                  "/${INSTALL_LIBDIR_RPM}/security")
 SET(INSTALL_PAMDATADIR_RPM              "/etc/security")
 
