--- mariadb-10.1.6/include/my_atomic.h.orig	2015-08-27 13:22:41.966316673 +0100
+++ mariadb-10.1.6/include/my_atomic.h	2015-08-27 13:25:20.225148975 +0100
@@ -328,7 +328,7 @@
 #define MY_ATOMIC_NOT_1CPU 1
 extern int my_atomic_initialize();
 
-#ifdef __ATOMIC_SEQ_CST
+#ifdef HAVE_GCC_C11_ATOMICS
 #define MY_MEMORY_ORDER_RELAXED __ATOMIC_RELAXED
 #define MY_MEMORY_ORDER_CONSUME __ATOMIC_CONSUME
 #define MY_MEMORY_ORDER_ACQUIRE __ATOMIC_ACQUIRE
