diff -up mariadb-10.3.6/storage/connect/javaconn.cpp.omv~ mariadb-10.3.6/storage/connect/javaconn.cpp
--- mariadb-10.3.6/storage/connect/javaconn.cpp.omv~	2018-05-08 12:27:06.032415657 +0200
+++ mariadb-10.3.6/storage/connect/javaconn.cpp	2018-05-08 12:27:29.989870590 +0200
@@ -153,7 +153,7 @@ bool JAVAConn::Check(jint rc)
 
 		if (exc != nullptr && tid != nullptr) {
 			jstring s = (jstring)env->CallObjectMethod(exc, tid);
-			const char *utf = env->GetStringUTFChars(s, (jboolean)false);
+			const char *utf = env->GetStringUTFChars(s, nullptr);
 			env->DeleteLocalRef(s);
 			Msg = PlugDup(m_G, utf);
 		} else
@@ -162,7 +162,7 @@ bool JAVAConn::Check(jint rc)
 		env->ExceptionClear();
 	} else if (rc < 0) {
 		s = (jstring)env->CallObjectMethod(job, errid);
-		Msg = (char*)env->GetStringUTFChars(s, (jboolean)false);
+		Msg = (char*)env->GetStringUTFChars(s, nullptr);
 	} else
 		Msg = NULL;
 
@@ -456,7 +456,7 @@ bool JAVAConn::Open(PGLOBAL g)
 
 		//=============== load and initialize Java VM and JNI interface =============
 		rc = CreateJavaVM(&jvm, (void**)&env, &vm_args);  // YES !!
-		delete options;    // we then no longer need the initialisation options.
+		delete[] options;    // we then no longer need the initialisation options.
 
 		switch (rc) {
 			case JNI_OK:
diff -up mariadb-10.3.6/storage/connect/jdbconn.cpp.omv~ mariadb-10.3.6/storage/connect/jdbconn.cpp
--- mariadb-10.3.6/storage/connect/jdbconn.cpp.omv~	2018-05-08 14:32:54.341228331 +0200
+++ mariadb-10.3.6/storage/connect/jdbconn.cpp	2018-05-08 14:33:34.085322405 +0200
@@ -836,11 +836,11 @@ bool JDBConn::Connect(PJPARM sop)
 		jstring s = (jstring)env->CallObjectMethod(job, qcid);
 
 		if (s != nullptr) {
-			char *qch = (char*)env->GetStringUTFChars(s, (jboolean)false);
+			char *qch = (char*)env->GetStringUTFChars(s, nullptr);
 			m_IDQuoteChar[0] = *qch;
 		} else {
 			s = (jstring)env->CallObjectMethod(job, errid);
-			Msg = (char*)env->GetStringUTFChars(s, (jboolean)false);
+			Msg = (char*)env->GetStringUTFChars(s, nullptr);
 		}	// endif s
 
 	}	// endif qcid
@@ -1018,7 +1018,7 @@ void JDBConn::SetColumnValue(int rank, P
 			cn = nullptr;
 
 		if (cn) {
-			field = env->GetStringUTFChars(cn, (jboolean)false);
+			field = env->GetStringUTFChars(cn, nullptr);
 			val->SetValue_psz((PSZ)field);
 		} else
 			val->Reset();
@@ -1092,7 +1092,7 @@ void JDBConn::SetColumnValue(int rank, P
 			cn = nullptr;
 
 		if (cn) {
-			const char *field = env->GetStringUTFChars(cn, (jboolean)false);
+			const char *field = env->GetStringUTFChars(cn, nullptr);
 			val->SetValue_psz((PSZ)field);
 		} else
 			val->Reset();
@@ -1470,7 +1470,7 @@ bool JDBConn::SetParam(JDBCCOL *colp)
 				return NULL;
 			} // endif label
 
-			name = env->GetStringUTFChars(label, (jboolean)false);
+			name = env->GetStringUTFChars(label, nullptr);
 			crp = qrp->Colresp;                    // Column_Name
 			crp->Kdata->SetValue((char*)name, i);
 			n = env->GetIntArrayElements(val, 0);
diff -up mariadb-10.3.6/storage/connect/jmgoconn.cpp.omv~ mariadb-10.3.6/storage/connect/jmgoconn.cpp
--- mariadb-10.3.6/storage/connect/jmgoconn.cpp.omv~	2018-05-08 14:42:27.874176972 +0200
+++ mariadb-10.3.6/storage/connect/jmgoconn.cpp	2018-05-08 14:42:58.212489349 +0200
@@ -272,7 +272,7 @@ bool JMgoConn::MakeCursor(PGLOBAL g, PTD
 
 			if (MakeSelector(g, filp, s)) {
 				strcpy(g->Message, "Failed making selector");
-				return NULL;
+				return false;
 			} else
 				s->Append('}');
 
@@ -340,7 +340,7 @@ bool JMgoConn::MakeCursor(PGLOBAL g, PTD
 
 				if (MakeSelector(g, filp, s)) {
 					strcpy(g->Message, "Failed making selector");
-					return NULL;
+					return false;
 				}	// endif Selector
 
 				tdbp->SetFilter(NULL);     // Not needed anymore
@@ -522,7 +522,7 @@ PSZ JMgoConn::GetDocument(void)
 		jdc = (jstring)env->CallObjectMethod(job, getdocid);
 
 		if (jdc)
-			doc = (PSZ)env->GetStringUTFChars(jdc, (jboolean)false);
+			doc = (PSZ)env->GetStringUTFChars(jdc, nullptr);
 
 	} // endif getdocid
 
@@ -807,7 +807,7 @@ PSZ JMgoConn::GetColumnValue(PSZ path)
 		fn = (jstring)env->CallObjectMethod(job, objfldid, jn);
 
 		if (fn)
-			fld = (PSZ)env->GetStringUTFChars(fn, (jboolean)false);
+			fld = (PSZ)env->GetStringUTFChars(fn, nullptr);
 
 	}	// endif objfldid
 
diff -up mariadb-10.3.6/storage/connect/tabjmg.cpp.omv~ mariadb-10.3.6/storage/connect/tabjmg.cpp
--- mariadb-10.3.6/storage/connect/tabjmg.cpp.omv~	2018-05-08 14:43:23.389918300 +0200
+++ mariadb-10.3.6/storage/connect/tabjmg.cpp	2018-05-08 14:43:30.079766481 +0200
@@ -101,7 +101,7 @@ bool JMGDISC::ColDesc(PGLOBAL g, jobject
 			continue;
 
 		jkey = (jstring)Jcp->env->CallObjectMethod(Jcp->job, bvnameid);
-		key = Jcp->env->GetStringUTFChars(jkey, (jboolean)false);
+		key = Jcp->env->GetStringUTFChars(jkey, nullptr);
 
 		if (pcn) {
 			strncpy(colname, pcn, 64);
