diff -up mariadb-10.10.4/storage/rocksdb/CMakeLists.txt.7~ mariadb-10.10.4/storage/rocksdb/CMakeLists.txt
--- mariadb-10.10.4/storage/rocksdb/CMakeLists.txt.7~	2023-05-12 19:44:57.801928373 +0200
+++ mariadb-10.10.4/storage/rocksdb/CMakeLists.txt	2023-05-12 19:45:35.012185272 +0200
@@ -35,12 +35,6 @@ ADD_DEFINITIONS(-Duint32_t=u_int32_t)
 ADD_DEFINITIONS(-Duint16_t=u_int16_t)
 ADD_DEFINITIONS(-Duint8_t=u_int8_t)
 
-# We've had our builders hang during the build process. This prevents MariaRocks
-# to be built on 32 bit intel OS kernels.
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "i[36]86")
-  SKIP_ROCKSDB_PLUGIN("Intel 32 bit not supported.")
-ENDIF()
-
 # Due to retrieved data being incorrect endian
 include(TestBigEndian)
 test_big_endian(BIG_ENDIAN)
