--- mariadb-10.1.11/storage/tokudb/ha_tokudb.cc.omv~	2016-02-01 13:44:25.227926030 +0100
+++ mariadb-10.1.11/storage/tokudb/ha_tokudb.cc	2016-02-01 13:44:38.762838642 +0100
@@ -288,7 +288,7 @@ static int free_share(TOKUDB_SHARE * sha
     }
 
 const char *ha_tokudb::table_type() const {
-    extern const char *tokudb_hton_name;
+    extern char const * const tokudb_hton_name;
     return tokudb_hton_name;
 } 
 
