diff -up mariadb-11.1.2/support-files/CMakeLists.txt.3~ mariadb-11.1.2/support-files/CMakeLists.txt
--- mariadb-11.1.2/support-files/CMakeLists.txt.3~	2023-08-17 17:25:34.000000000 +0200
+++ mariadb-11.1.2/support-files/CMakeLists.txt	2023-08-22 11:54:32.210117534 +0200
@@ -237,6 +237,7 @@ IF(UNIX AND NOT WITHOUT_SERVER)
               COMPONENT SharedLibraries)
       INSTALL(FILES rpm/mysql-clients.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT Client)
+      CONFIGURE_FILE(rpm/server.cnf ${CMAKE_CURRENT_SOURCE_DIR}/rpm/server.cnf @ONLY)
       INSTALL(FILES rpm/server.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT IniFiles)
       INSTALL(FILES rpm/enable_encryption.preset DESTINATION ${INSTALL_SYSCONF2DIR}
diff -up mariadb-11.1.2/support-files/rpm/server.cnf.3~ mariadb-11.1.2/support-files/rpm/server.cnf
--- mariadb-11.1.2/support-files/rpm/server.cnf.3~	2023-08-22 11:54:32.210117534 +0200
+++ mariadb-11.1.2/support-files/rpm/server.cnf	2023-08-22 11:55:55.238924829 +0200
@@ -13,6 +13,10 @@
 
 # This group is read by both MariaDB and MySQL servers
 [mysqld]
+datadir=@MYSQL_DATADIR@
+socket=@MYSQL_UNIX_ADDR@
+log-error=@LOG_LOCATION@
+pid-file=@PID_FILE_DIR@/@DAEMON_NO_PREFIX@.pid
 
 #
 # * Galera-related settings
