--- mariadb-10.1.5/storage/tokudb/ft-index/tools/tokudb_dump.cc.omv~	2015-07-15 12:13:12.754830835 +0200
+++ mariadb-10.1.5/storage/tokudb/ft-index/tools/tokudb_dump.cc	2015-07-15 12:13:40.303733615 +0200
@@ -247,6 +247,7 @@ outputplaintextstring(char* str)
    g.plaintext = old_plaintext;
 }
 
+#if 0
 static inline int
 hextoint(int ch)
 {
@@ -316,6 +317,7 @@ error:
    PRINT_ERROR(0, "Quitting out due to errors.\n");
    return EXIT_FAILURE;
 }
+#endif
 
 static inline int
 verify_library_version(void)
