diff -up mariadb-10.7.1/mysql-test/main/ssl_cipher.test.5~ mariadb-10.7.1/mysql-test/main/ssl_cipher.test
--- mariadb-10.7.1/mysql-test/main/ssl_cipher.test.5~	2021-12-03 21:02:11.160857155 +0100
+++ mariadb-10.7.1/mysql-test/main/ssl_cipher.test	2021-12-03 21:03:10.072133417 +0100
@@ -96,7 +96,9 @@ drop user mysqltest_1@localhost;
 #
 let $restart_parameters=--ssl-cipher=AES128-SHA;
 source include/restart_mysqld.inc;
+--replace_regex /TLS_AES_.*/AES128-SHA/
 connect (ssl_con,localhost,root,,,,,SSL);
+--replace_regex /TLS_AES_.*/AES128-SHA/
 SHOW STATUS LIKE 'Ssl_cipher';
 SELECT VARIABLE_VALUE like '%AES128-SHA%' FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME='Ssl_cipher_list';
 disconnect ssl_con;
