diff -up mariadb-12.3.3/support-files/mariadb.logrotate.sh.1~ mariadb-12.3.3/support-files/mariadb.logrotate.sh
--- mariadb-12.3.3/support-files/mariadb.logrotate.sh.1~	2026-08-25 12:00:00.000000000 +0000
+++ mariadb-12.3.3/support-files/mariadb.logrotate.sh	2026-08-25 12:00:00.000000000 +0000
@@ -6,7 +6,8 @@
 # Read https://mariadb.com/docs/server/server-management/server-monitoring-logs/error-log to learn more about logging and
 # https://mariadb.com/docs/server/server-management/server-monitoring-logs/rotating-logs-on-unix-and-linux about rotating logs.
 
-@localstatedir@/mysqld.log @localstatedir@/mariadb.log @logdir@/*.log {
+@LOG_LOCATION@ @localstatedir@/mariadb.log @logdir@/*.log {
+  create 600 mysql mysql
 
   # Depends on a mysql@localhost unix_socket authenticated user with RELOAD privilege
   @su_user@
