--- mariadb-10.1.22/scripts/CMakeLists.txt.orig	2017-04-23 09:36:36.593968137 +0100
+++ mariadb-10.1.22/scripts/CMakeLists.txt	2017-04-23 09:37:12.659574135 +0100
@@ -252,6 +252,34 @@
   DESTINATION ${INSTALL_SCRIPTDIR}
   COMPONENT Server
   )
+
+  # files for systemd
+  SET(SYSTEMD_INST_SCRIPTS
+    mysql.tmpfiles.d
+    mysql.service
+    mysql@.service
+    mysql-prepare-db-dir
+    mysql-wait-ready
+    mysql-wait-stop
+    mysql-check-socket
+    mysql-check-upgrade
+    mysql-scripts-common
+  )
+  FOREACH(sysdfile ${SYSTEMD_INST_SCRIPTS})
+    SET(INFILE ${CMAKE_CURRENT_SOURCE_DIR}/${sysdfile}.in)
+    SET(SHFILE ${CMAKE_CURRENT_SOURCE_DIR}/${sysdfile}.sh)
+    IF(EXISTS ${SHFILE})
+      CONFIGURE_FILE(${SHFILE}
+        ${CMAKE_CURRENT_BINARY_DIR}/${sysdfile} ESCAPE_QUOTES @ONLY)
+    ELSEIF(EXISTS ${INFILE})
+      CONFIGURE_FILE(${INFILE}
+        ${CMAKE_CURRENT_BINARY_DIR}/${sysdfile} ESCAPE_QUOTES @ONLY)
+    ELSE()
+      MESSAGE(FATAL_ERROR "Can not find ${SHFILE} or ${INFILE} in "
+        "${CMAKE_CURRENT_SOURCE_DIR}" )
+    ENDIF()
+  ENDFOREACH()
+
 ENDIF()
 
 SET(prefix "${CMAKE_INSTALL_PREFIX}")
