--- mariadb-10.0.14/storage/connect/filamap.h.omv~	2014-10-09 07:38:02.631589782 +0200
+++ mariadb-10.0.14/storage/connect/filamap.h	2014-10-09 07:39:02.178625211 +0200
@@ -104,7 +104,7 @@ class DllExport MPXFAM : public MBKFAM {
   virtual int   MaxBlkSize(PGLOBAL g, int s)
                 {return TXTFAM::MaxBlkSize(g, s);}
   virtual bool  SetPos(PGLOBAL g, int recpos);
-  virtual int   GetNextPos(void) {return (int)Fpos + Nrec;}
+  virtual int   GetNextPos(void) {return (int)(intptr_t)Fpos + Nrec;}
   virtual bool  DeferReading(void) {return false;}
   virtual int   ReadBuffer(PGLOBAL g);
   virtual int   WriteBuffer(PGLOBAL g);
--- mariadb-10.0.14/storage/tokudb/ft-index/cmake_modules/TokuSetupCompiler.cmake.omv~	2014-10-09 07:44:27.823975199 +0200
+++ mariadb-10.0.14/storage/tokudb/ft-index/cmake_modules/TokuSetupCompiler.cmake	2014-10-09 07:44:45.947420767 +0200
@@ -19,6 +19,7 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL FreeB
   add_c_defines(
     _SVID_SOURCE
     _XOPEN_SOURCE=600
+    _DEFAULT_SOURCE=1
     )
 endif ()
 
--- mariadb-10.0.14/storage/tokudb/ft-index/ft/txn/rollback.cc.omv~	2014-10-09 07:46:53.021435648 +0200
+++ mariadb-10.0.14/storage/tokudb/ft-index/ft/txn/rollback.cc	2014-10-09 07:47:04.431493321 +0200
@@ -145,9 +145,7 @@ static inline PAIR_ATTR make_rollback_pa
 PAIR_ATTR
 rollback_memory_size(ROLLBACK_LOG_NODE log) {
     size_t size = sizeof(*log);
-    if (&log->rollentry_arena) {
-        size += log->rollentry_arena.total_footprint();
-    }
+    size += log->rollentry_arena.total_footprint();
     return make_rollback_pair_attr(size);
 }
 
