--- mariadb-10.1.5/storage/tokudb/PerconaFT/tools/tokudb_dump.cc.omv~	2015-07-15 12:13:12.754830835 +0200
+++ mariadb-10.1.5/storage/tokudb/PerconaFT/tools/tokudb_dump.cc	2015-07-15 12:13:40.303733615 +0200
@@ -247,6 +247,7 @@ outputplaintextstring(char* str)
    g.plaintext = old_plaintext;
 }
 
+#if 0
 static inline int
 hextoint(int ch)
 {
@@ -316,6 +317,7 @@ error:
    PRINT_ERROR(0, "Quitting out due to errors.\n");
    return EXIT_FAILURE;
 }
+#endif
 
 static inline int
 verify_library_version(void)
--- ./storage/tokudb/PerconaFT/ft/logger/recover.cc.orig 2015-09-13 12:43:58.882978585 +0100
+++ ./storage/tokudb/PerconaFT/ft/logger/recover.cc      2015-09-13 12:44:29.145761519 +0100
@@ -785,7 +785,7 @@
     assert(txn!=NULL);
 
     // commit the transaction
-    toku_txn_progress_extra extra = { time(NULL), l->lsn, "commit", l->xid };
+    toku_txn_progress_extra extra = { time(NULL), l->lsn, "commit", l->xid, 0 };
     int r = toku_txn_commit_with_lsn(txn, true, l->lsn, toku_recover_txn_progress, &extra);
     assert(r == 0);
 
@@ -828,7 +828,7 @@
     assert(txn!=NULL);
 
     // abort the transaction
-    toku_txn_progress_extra extra = { time(NULL), l->lsn, "abort", l->xid };
+    toku_txn_progress_extra extra = { time(NULL), l->lsn, "abort", l->xid, 0 };
     r = toku_txn_abort_with_lsn(txn, l->lsn, toku_recover_txn_progress, &extra);
     assert(r == 0);
 
@@ -1363,7 +1363,7 @@
     // sanity check that the recursive call successfully NULLs out txn->child
     invariant(txn->child == NULL);
     // abort the transaction
-    toku_txn_progress_extra extra = { time(NULL), ZERO_LSN, "abort live", txn->txnid };
+    toku_txn_progress_extra extra = { time(NULL), ZERO_LSN, "abort live", txn->txnid, 0 };
     int r = toku_txn_abort_txn(txn, toku_recover_txn_progress, &extra);
     assert(r == 0);
     
