--- mariadb-10.1.14/libmysql/CMakeLists.txt.0004~	2016-05-10 20:34:43.516900774 +0200
+++ mariadb-10.1.14/libmysql/CMakeLists.txt	2016-05-10 20:37:26.711867096 +0200
@@ -422,7 +422,7 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
     @ONLY
   )
   SET(VERSION_SCRIPT_LINK_FLAGS 
-    "-Wl,${CMAKE_CURRENT_BINARY_DIR}/libmysql_versions.ld")
+    "-fuse-ld=bfd -Wl,${CMAKE_CURRENT_BINARY_DIR}/libmysql_versions.ld")
 
 ENDIF(CMAKE_SYSTEM_NAME MATCHES "Linux")
 
