--- mariadb-10.1.1/scripts/mysql_config.pl.in.omv~	2014-11-06 14:02:07.096476066 +0100
+++ mariadb-10.1.1/scripts/mysql_config.pl.in	2014-11-06 14:02:09.875441324 +0100
@@ -43,7 +43,7 @@ use strict;
 
 my @exclude_cflags =
   qw/DDBUG_OFF DSAFE_MUTEX DUNIV_MUST_NOT_INLINE DFORCE_INIT_OF_VARS
-     DEXTRA_DEBUG DHAVE_valgrind O O[0-9] xO[0-9] W[-A-Za-z]*
+     DEXTRA_DEBUG DHAVE_valgrind O O[0-9] xO[0-9]
      Xa xstrconst xc99=none
      unroll2 ip mp restrict/;
 
