diff --git a/mysys_ssl/my_md5.cc b/mysys_ssl/my_md5.cc
index 7139ea9..0a61f84 100644
--- a/mysys_ssl/my_md5.cc
+++ b/mysys_ssl/my_md5.cc
@@ -83,7 +83,11 @@ static void md5_input(MD5_CONTEXT *context, const uchar *buf, unsigned len)
 static void md5_result(MD5_CONTEXT *context, uchar digest[MD5_HASH_SIZE])
 {
   EVP_DigestFinal_ex(context, digest, NULL);
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
   EVP_MD_CTX_cleanup(context);
+#else
+  EVP_MD_CTX_reset(context);
+#endif
 }
 
 #endif /* HAVE_YASSL */
